modules/virt-creating-single-pvc-template-storage-pool.adoc

```diff
@@ -0,0 +1,54 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/virtual_disks/virt-configuring-local-storage-for-vms.adoc
+
+:_content-type: PROCEDURE
+[id="virt-creating-single-pvc-template-storage-pool_{context}"]
+= Creating a single `PVCTemplate` storage pool in a host path provisioner custom resource
+
+Specifying a single `pvcTemplate` (a standard Kubernetes `PersistentVolumeClaimSpec`)  allows you to specify a persistent volume claim (PVC) template, which is mounted on the node at the where you want to create hostpath provisioner (HPP) volumes. Using single `pvcTemplate`s can improve I/O performance. By configuring a cluster where each node has an additional disk to store the HPP, you effectively distribute many large volumes over multiple smaller virtual volumes.
+
+[NOTE]
+====
+You can use the Local Storage Operator to create provisioned volumes that can used by the HPP.
+====
+
+.Procedure
+
+. Create a YAML file for the HPP CSI driver provisioner custom resource specifying a single `pvcTemplate`. For example:
++
+[source,terminal]
+----
+$ touch hostpathprovisioner_cr_singlepvc.yaml
+----
+
+. Edit the file. For example:
++
+[source,yaml]
+----
+apiVersion: hostpathprovisioner.kubevirt.io/v1beta1
+kind: HostPathProvisioner
+metadata:
+ name: hostpath-provisioner
+spec:
+ imagePullPolicy: IfNotPresent
+ storagePools:
+   - name: local
+  pvcTemplate:
+       volumeMode: Block <1>
+       storageClassName: lso <2>
+       accessModes:
+       - ReadWriteOnce
+       resources:
+         requests:
+           storage: 5Gi <3>
+     path: "</var/myvolumes>"
+ workload:
+   nodeSelector:
+     kubernetes.io/os: linux
+----
+<1> `volumeMode` can be `Block` or `Filesystem`, but must match the provisioned volume format. Default is `Filesystem`. Note that the Local Storage Operator (LSO) writes only in `block` format. If the `volumeMode` is `block`, then the mounting pod creates an XFS filesystem on the block volume before mounting it.
+<2> If the `storageClassName` is omitted, the default storage class is used to create PVCs. If you omit `storageClassName`, ensure that the HPP storage class is not the default storage class.
```

Comment:
We can probably make this simpler by noting that the entire pvcTemplate is a kubernetes PVC spec object, and follows the exact same rules. so volumeMode, accessMode, storageClass and size are all the same any other PVC yaml. We should probably phrase that better, but my point is, nothing in here is new.

Reply:
Resolved